docs/source/tutorials/token_classification/utils.py

```diff
@@ -0,0 +1,242 @@
+        while acc != len(token): 
+            token_len = len(tokens[i_new][j_new:]) 
+            remain = len(token) - acc 
+            weights.append(min(remain, token_len)) 
+            if token_len > remain: 
+                acc += remain 
+                j_new += remain 
+            else: 
+                acc += token_len 
+                i_new += 1 
+                j_new = 0 
+        
+        if i != i_new: 
+            probs = np.average(scores[i:i_new], axis=0, weights=weights if weighted else None) 
+        else: 
+            probs = scores[i] 
+        i, j = i_new, j_new 
+        
+        pred_probs.append(probs) 
+        labels.append(label)
+        
+    return np.array(pred_probs), labels 
+
+def to_dict(nl): 
+    return {str(i): l for i, l in enumerate(nl)} 
+
+def read_npz(filepath): 
+    data = dict(np.load(filepath)) 
+    data = [data[str(i)] for i in range(len(data))] 
+    return data 
+
+def color_sentence(sentence, word): 
+    start_idx = sentence.index(word) 
+    before, after = sentence[:start_idx], sentence[start_idx + len(word):]
+    return '%s%s%s' % (before, colored(word, 'red'), after) 
+
+
+def show_issues(issues, labels, pred_probs, given_words, sentences, 
+                top=10, entities=None, exclude=[]): 
+    if not entities: 
+        entities = ['O', 'MISC', 'PER', 'ORG', 'LOC'] 
+    def show_issue(issue): 
+        i, j = issue 
+        issue_label = labels[i][j] 
+        predicted_label = np.argmax(pred_probs[i][j]) 
+        issue_word = given_words[i][j] 
+        issue_sentence = color_sentence(sentences[i], issue_word) 
+        return issue_sentence, (issue_label, predicted_label)  
+    
+    shown = 0 
+    for issue in issues: 
+        sentence, (given, predicted) = show_issue(issue) 
+        if (given, predicted) in exclude: 
+            continue 
+        print('%d. %s' % (shown+1, sentence)) 
+        print('Given label: %s, predicted label: %s\n' % (entities[given], entities[predicted])) 
+        shown += 1 
+        if shown == top: 
+            break 
+
+def common_token_issues(issues, given_words, labels, pred_probs, entities, 
+                        top=10, exclude=[], verbose=True): 
+    n = pred_probs[0].shape[1] 
+    count = {} 
+    for issue in issues: 
+        i, j = issue 
+        word = given_words[i][j] 
+        label = labels[i][j] 
+        pred = pred_probs[i][j].argmax() 
+        if word not in count: 
+            count[word] = np.zeros([n, n], dtype=int) 
+        if (label, pred) not in exclude: 
+            count[word][label][pred] += 1 
+    words = [word for word in count.keys()] 
+    n = count[words[0]].shape[0] 
+    freq = [np.sum(count[word]) for word in words] 
+    rank = np.argsort(freq)[::-1][:top] 
+    
+    for r in rank: 
+        word = words[r] 
+        matrix = count[word] 
+        most_frequent = np.argsort(matrix.flatten())[::-1] 
+        print("'%s' is mislabeled %d times" % (word, freq[r])) 
```

Comment:
Let's change the print statement to read:

Token 'XYZ' is potentially mislabeled in J sentences

instead of "Token XYZ is mislabeled J times"

Reply:
edge case: there may be one sentence with two label issues with the exact same token (although it happens really rarely)

Reply:
This way if the user does not supply the strings corresponding to each token, this function could still print out:

Token 1394 is potentially mislabeled in J sentences

where say 1394 would correspond to the index of token 'XYZ' in the vocabulary.  I think that way you can make `given_words = None` an optional arg.  But let's include print statement inside the function:

```
if given_words is None:
   print("Tokens will be printed in terms of their integer index since `given_words` was not provided, specify this argument to see the string names of each token").
```

Reply:
Let's also move this print inside of the verbose conditional, so this function prints nothing if `verbose=False`, it just returns the dataframe mentioned in comment below.